### PR TITLE
Improve Rust syntax highlighting

### LINKED ISSUE #56: Rust function definitions
Hi,

I’ve noticed that function definitions in most languages seem to be highlighted in orange (`Title` highlight group?). Would you be open to linking `rustFuncName` to `Title` to make this more consistent?

Thanks for such a nice colourscheme 👍

### PR DESCRIPTION
**Pull request by gkeep**
Resolves #56 and generally improves Rust highlighting.

Code snippet more or less showcasing the changes:
Code snippet

### CODE REVIEW

**Comment by cocopon** [comment]
Sorry for the late reply... Could you share the example code in the screenshot? I want to check it in my environment.

**Comment by gkeep** [comment]
Sure. [Here](https://hasteb.in/alenohew.cpp) it is.

**Comment by cocopon** [comment]
Thank you!

- Link from `rustFuncName` to `Title` looks nice.
- Link from `rustFuncCall` to `Special` is much green so I think it should be removed. Green color is for Special keyword (e.g. type notation in TypeScript, `TODO:` mark, etc.). The important thing is not colorfulness but consistency and readability.
- Do you have any example that contains `rustTrait` and `rustType`? I also want to check them in actual code.

**Comment by gkeep** [comment]
Thank you for feedback! 

Sadly, I cannot work on the PR at the moment (on a vacation), but I can talk with you on some your points :)

> Link from rustFuncCall to Special is much green so I think it should be removed. Green color is for Special keyword (e.g. type notation in TypeScript, TODO: mark, etc.). The important thing is not colorfulness but consistency and readability.

I took python's syntax highlighting as an example, where functions like ’var.append()’ (function append) were highlighted with green. 


> Do you have any example that contains rustTrait and rustType? I also want to check them in actual code.

As far as I remember, you can check ’rustType’ by declaring a variable with a type:
’let var: String;’

I think ‘rustTrait’ appears when you declare variables in the head of a function. I had it in the earlier version of code, I will look for it (or just make the repo public)...

Sorry for the formatting, writing on a phone.

**Comment by cocopon** [comment]
> I cannot work on the PR at the moment (on a vacation)

I hope you enjoy your vacation ;)

> I took python's syntax highlighting as an example, where functions like var.append() (function append) were highlighted with green.

I cannot find `.append()` that is green. Are you using any additional syntax plugin for Python?



> `rustType`, `rustTrait`

Thank you! I found them. `rustTrait` is linked to `rustType` by default so you don't have to link them explicitly.

**Comment by gkeep** [comment]
> I hope you enjoy your vacation ;)

Thanks, that’s very kind of you. 

> Are you using any additional syntax plugin for Python?

I do, actually, use [ALE](https://github.com/dense-analysis/ale) as a linting engine. Never made the connection, though.

After disabling ALE, functions are not highlighted with green. 


> `rustTrait` is linked to `rustType` by default so you don't have to link them explicitly.

They were not linked for me, but again, it might have been because of ALE.

File: colors/iceberg.vim [hunk 1]
```diff
@@ -133,6 +133,10 @@ if &background == 'light'
   hi! link rubyInterpolationDelimiter String
   hi! link rubySharpBang Comment
   hi! link rubyStringDelimiter String
+  hi! link rustFuncCall Special
+  hi! link rustFuncName Title
+  hi! link rustTrait Constant
```
Comment (cocopon) [inline colors/iceberg.vim:138]: `rustTrait` is linked to `rustType` by default so you can remove this line.
https://github.com/vim/vim/blob/b146e01a7e41bf5787a2b5ccc6c340ed6001f308/runtime/syntax/rust.vim#L219

File: colors/iceberg.vim [hunk 2]
```diff
@@ -363,6 +367,10 @@ else
...
```
Comment (cocopon) [inline colors/iceberg.vim:372]: Same as https://github.com/cocopon/iceberg.vim/pull/67/files#diff-1e50a772bd2d4e5eb101b7a73326a862R138

**Comment by gkeep** [comment]
image

This is how it looks with the changes you requested. (Thank you SSH)

I'd push the commits now, but I’m double checking about coloring of function calls. 

In Python they are not highlighted (same as text/white), here they are highlighted with cyan.

**Comment by cocopon** [comment]
Thank you! It looks better than before.

I also want to add `hi! link rustFuncCall Normal` to suppress excessive syntax highlighting. I think it helps to focus important words. What do you think?

Current:


Linking `rustFuncCall` to `Normal`:

**Comment by gkeep** [comment]
I agree, that looks way better than before.

Current version (blue) looks really monotonous, since namespaces are highlighted with the same color. 

I could argue about using green as a color for function calls, but I see your point on readability, not colorfulness.

Anyway, the PR should be ready for merging soon.

**Comment by cocopon** [comment]
Merged. Thanks!

**Pull request merged**